drift_dev/lib/src/model/sql_query.dart

```diff
@@ -443,19 +468,67 @@ class NestedResultTable {
   String get dartFieldName => ReCase(name).camelCase;
 
   /// [hashCode] that matches [isCompatibleTo] instead of `==`.
+  @override
   int get compatibilityHashCode {
     return Object.hash(name, table);
   }
 
   /// Checks whether this is compatible to the [other] nested result, which is
   /// the case iff they have the same and read from the same table.
-  bool isCompatibleTo(NestedResultTable other) {
+  @override
+  bool isCompatibleTo(NestedResult other) {
+    if (other is! NestedResultTable) return false;
+
     return other.name == name &&
         other.table == table &&
         other.isNullable == isNullable;
   }
 }
 
+class NestedResultQuery extends NestedResult {
+  final NestedQueryColumn from;
+
+  final SqlSelectQuery query;
+
+  NestedResultQuery({
+    required this.from,
+    required this.query,
+  });
+
+  String filedName(String? nestedPrefix) {
+    if (from.as != null) {
+      return from.as!;
+    }
+
+    final name = ReCase(query.resultClassName).camelCase;
+
+    if (nestedPrefix != null) {
+      return name + nestedPrefix;
+    }
+
+    return name;
+  }
+
+  String resultTypeCode(String parentResultClassName) {
+    if (query.resultSet.needsOwnClass) {
+      return parentResultClassName + query.resultClassName;
+    } else {
+      return query.resultTypeCode();
+    }
+  }
+
+  // Every query should be unique.
+
+  /// [hashCode] that matches [isCompatibleTo] instead of `==`.
+  @override
+  int get compatibilityHashCode => hashCode;
+
+  /// Checks whether this is compatible to the [other] nested result, which is
+  /// the case iff they have the same and read from the same table.
+  @override
+  bool isCompatibleTo(NestedResult other) => this == other;
+}
```

Comment:
Maybe the first comment should also point out that we don't currently support using nested queries with reusable [result class name](https://drift.simonbinder.eu/docs/using-sql/drift_files/#result-class-names), which is the reason those methods exist.

IMO the documentation comments don't add much value on this subclass.

Reply:
Yes I agree. I probably just left them in because I copied these methods from the super class.